### ISSUE TITLE
Azure?

### ISSUE THREAD

**Issue by chadbr** [Microsoft/Microsoft.Diagnostics.Tracing.Logging#1]
Is there any examples of using this library in Azure? We're trying to use it in Azure websites / Azure web apps and we'd like to turn on ETW tracing programmatically -- we haven't found any solutions?

Thanks

**Comment by doubleyewdee** [comment]
I don't have a ready example for you but I'd love to work with you on Azure integrated. ETW has some peculiar requirements for use which may or may not be easily solved with Azure services.

That said for intra process logging this library will work cleanly in pretty much any modern Windows environment.

**Comment by chadbr** [comment]
Yes - we have used it extensively in our Windows apps. We were looking for a way to control logging at runtime by switching it on and off from .net. In Windows this works fine because we can control the security. In Azure Web Apps - we don't see a way to elevate the permissions to start logging.

We were wondering if there was a way to elevate the permissions or if there is some way to do it through the Azure portal so we could enable / disable logging at runtime. Otherwise we'll have to push all out custom logging types down to a flat structure and send it through simple tracing (and then we can't use all the custom viewers we've built, etc...)

This may be out of scope -- but it there is a way to get your library turning on and off through Azure web apps - then we should be able to do the same for ours.

Thanks - Chad

**Comment by doubleyewdee** [comment]
Right now "sort of." It can detect automatically whether you can create real ETW sessions and switch to text logging if you can't. But that means only intra-process messages are logged. If you want providers that are not EventSources within your app it's not really possible.

There are also user mode ETW sessions which carry similar but not identical limits. What is your end goal?

**Comment by chadbr** [comment]
End game is:

Either with a website, a remote WPF console, or command line app - enable ETW logging in production. We have several custom EventSource's we'd like to capture. 

I assume we'll have to do something like ETW -> web socket relay of some sort (listener). 

How we get there I'm flexible on - looking for ideas.
